### PR TITLE
Remove More Naked GL

### PR DESCRIPTION
**Pull request by ZappoMan**
In an effort to clean up our cross platform rendering pipeline we are removing all old uses of nake OpenGL code. This removes naked GL from the following items:

- MyAvatar::renderDebugBodyPoints() which was not being called.
- fixed SkeletonModel::renderJointConstraints()
- fixed ImageOverlays
- Removed AudioIOStatsRenderer which was not being called.
- Removed Util.cpp/drawText() which was only being used by AudioIOStatsRenderer

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5a927c4c90829fd312cd9428dc73b69376a4c78d available at: https://gist.github.com/db25c929159696e77fb2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5330.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5330.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8bb6616a2587ddd815063965f683ce5ed8580ea3 available at: https://gist.github.com/db25c929159696e77fb2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5330.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5330.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5d43c98ebc1c1c19a96b7d8d0550dfc95756caf0 available at: https://gist.github.com/db25c929159696e77fb2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5330.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5330.dmg

File: interface/src/Menu.cpp [hunk 1]
```diff
@@ -588,18 +587,13 @@ Menu::Menu() {
         audioScopeFramesGroup->addAction(fiftyFrames);
     }
 
-    auto statsRenderer = DependencyManager::get<AudioIOStatsRenderer>();
     addCheckableActionToQMenuAndActionHash(audioDebugMenu, MenuOption::AudioStats,
                                            Qt::CTRL | Qt::SHIFT | Qt::Key_A,
-                                           false,
-                                           statsRenderer.data(),
-                                           SLOT(toggle()));
+                                           false); //, statsRenderer.data(), SLOT(toggle())); // TODO: convert to dialogbox
```
Comment (ZappoMan) [inline interface/src/Menu.cpp]: This is going to get converted to a dialog box soon.

File: interface/src/Menu.cpp [hunk 2]
```diff
@@ -588,18 +587,13 @@ Menu::Menu() {
...
 
     addCheckableActionToQMenuAndActionHash(audioDebugMenu, MenuOption::AudioStatsShowInjectedStreams,
                                             0,
-                                            false,
-                                            statsRenderer.data(),
-                                            SLOT(toggleShowInjectedStreams()));
+                                           false); //, statsRenderer.data(), SLOT(toggleShowInjectedStreams)); // TODO: convert to dialogbox
```
Comment (ZappoMan) [inline interface/src/Menu.cpp]: This is going to get converted to a dialog box soon.

File: interface/src/Util.cpp [hunk 3]
```diff
@@ -100,23 +100,6 @@ int widthText(float scale, int mono, char const* string) {
     return textRenderer(mono)->computeExtent(string).x;  // computeWidth(string) * (scale / 0.10);
 }
 
-void drawText(int x, int y, float scale, float radians, int mono,
```
Comment (ZappoMan) [inline interface/src/Util.cpp]: was only being used by AudioIOStatsRenderer

File: interface/src/audio/AudioIOStatsRenderer.cpp [hunk 4]
```diff
@@ -1,242 +0,0 @@
-//
-//  AudioIOStatsRenderer.cpp
```
Comment (ZappoMan) [inline interface/src/audio/AudioIOStatsRenderer.cpp]: was not being called at all

File: interface/src/avatar/MyAvatar.cpp [hunk 5]
```diff
@@ -323,28 +323,6 @@ void MyAvatar::updateFromTrackers(float deltaTime) {
 }
 
 
-void MyAvatar::renderDebugBodyPoints() {
```
Comment (ZappoMan) [inline interface/src/avatar/MyAvatar.cpp]: was not being called at all

**Comment by hifi-gustavo** [comment]
Build console output for commit: e39ea72daf96604ff215547d5818f1f16e2cd3cc available at: https://gist.github.com/db25c929159696e77fb2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5330.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5330.dmg

File: interface/src/avatar/SkeletonModel.cpp [hunk 6]
```diff
@@ -362,7 +363,7 @@ void SkeletonModel::renderJointConstraints(int jointIndex) {
             // TODO: this is really inefficient constantly recreating these vertices buffers. It would be
             // better if the skeleton model cached these buffers for each of the joints they are rendering
             geometryCache->updateVertices(_triangleFanID, points, color);
-            geometryCache->renderVertices(gpu::TRIANGLE_FAN, _triangleFanID);
+            geometryCache->renderVertices(batch, gpu::TRIANGLE_FAN, _triangleFanID);
             
         }
         glPopMatrix();
```
Comment (jherico) [inline interface/src/avatar/SkeletonModel.cpp]: missed a spot.

File: interface/src/ui/overlays/ImageOverlay.cpp [hunk 7]
```diff
@@ -62,42 +62,18 @@ void ImageOverlay::render(RenderArgs* args) {
...
-
-        gpu::StatePointer blitState = gpu::StatePointer(new gpu::State());
-        blitState->setBlendFunction(false, gpu::State::SRC_ALPHA, gpu::State::BLEND_OP_ADD, gpu::State::INV_SRC_ALPHA);
-        blitState->setColorWriteMask(true, true, true, true);
-        drawPipeline = gpu::PipelinePointer(gpu::Pipeline::create(blitProgram, blitState));
-    }
-    */
-    // TODO: batch.setPipeline(drawPipeline);
-    glUseProgram(0);
+    gpu::Batch& batch = *args->_batch;
```
Comment (jherico) [inline interface/src/ui/overlays/ImageOverlay.cpp]: I'm not sure there's always a batch here when the overlays are rendered.  do we have a test for rendering all the types of overlay as an example?

Comment (ZappoMan) [inline interface/src/ui/overlays/ImageOverlay.cpp]: I tested that the overlays work. The other overlays assume a batch is always available.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 53c3e3d7310265111b6e6533a291c0f64509fe92 available at: https://gist.github.com/db25c929159696e77fb2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5330.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5330.dmg

**Pull request merged**